meson: Could not run pari version

### Environment

- **OS**: macOS Tahoe 26.4 (almost out of the box)
- **Sage Version**: 10.9.beta8 


### Steps To Reproduce

At the `pip install --no-build-isolation --editable .` step, I get the following error:

```
(sage-dev) a549@549s-MacBook-Pro sage % pip install --no-build-isolation --editable .     
Obtaining file:///Users/a549/programs/sage
  Checking if build backend supports build_editable ... done
  Preparing editable metadata (pyproject.toml) ... error
  error: subprocess-exited-with-error
  
  × Preparing editable metadata (pyproject.toml) did not run successfully.
  │ exit code: 1
  ╰─> [24 lines of output]
      + meson setup --reconfigure /Users/a549/programs/sage /Users/a549/programs/sage/build/cp314 -Dbuildtype=release -Db_ndebug=if-release -Db_vscrt=md --default-library=static --native-file=/Users/a549/programs/sage/build/cp314/meson-python-native-file.ini
      The Meson build system
      Version: 1.10.1
      Source dir: /Users/a549/programs/sage
      Build dir: /Users/a549/programs/sage/build/cp314
      Build type: native build
      Project name: SageMath
      Project version: 10.9.beta8
      C compiler for the host machine: arm64-apple-darwin20.0.0-clang (clang 19.1.7 "clang version 19.1.7")
      C linker for the host machine: arm64-apple-darwin20.0.0-clang ld64 956.6
      C++ compiler for the host machine: arm64-apple-darwin20.0.0-clang++ (clang 19.1.7 "clang version 19.1.7")
      C++ linker for the host machine: arm64-apple-darwin20.0.0-clang++ ld64 956.6
      Cython compiler for the host machine: cython (cython 3.2.4)
      Host machine cpu family: aarch64
      Host machine cpu: aarch64
      Program python found: YES (/Users/a549/mamba/envs/sage-dev/bin/python3.14)
      Program python (pytest) found: YES (/Users/a549/mamba/envs/sage-dev/bin/python3.14) modules: pytest
      Dependency numpy found: YES 2.4.2 (cached)
      Dependency gmp found: YES 6.3.0 (cached)
      Library pari found: YES
      
      ../../src/meson.build:102:22: ERROR: Could not run pari version
      
      A full log can be found at /Users/a549/programs/sage/build/cp314/meson-logs/meson-log.txt
      [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
error: metadata-generation-failed

× Encountered error while generating package metadata.
╰─> from file:///Users/a549/programs/sage

note: This is an issue with the package mentioned above, not pip.
hint: See above for details.
```

### Meson log
[meson-log.txt](https://github.com/user-attachments/files/26367163/meson-log.txt)

### Additional Information

Commenting out the `cc.run` and the check (around `src/meson.build:102`) for the pari version allows meson to proceed normally. I can also build the C code that is intended to output the version and that also works.

### Checklist

- [x] I have searched the existing issues for a bug report that matches the one I want to file, without success.
- [x] I have read the documentation and troubleshoot guide

I reproduce this today when compiling the newest repo

My envs:
- macOS arm64 26.3.1 (a) 
- `sage-dev` conda environment (environment-3.13-macos.yml)
- compiler used by Meson: `arm64-apple-darwin20.0.0-clang` 19.1.7

```bash
echo "$CONDA_PREFIX"
which python
which pkg-config
which arm64-apple-darwin20.0.0-clang
python -V
---
/Users/xxx/miniforge3/envs/sage-dev
/Users/xxx/miniforge3/envs/sage-dev/bin/python
/Users/xxx/miniforge3/envs/sage-dev/bin/pkg-config
/Users/xxx/miniforge3/envs/sage-dev/bin/arm64-apple-darwin20.0.0-clang
Python 3.13.11
```

During build
```bash
pip install --no-build-isolation --no-deps --config-settings=builddir=builddir --editable . -v
```

Meson first report `Library pari found: YES`, then fails at the PARI version probe with `ERROR: Could not run pari version`
(Same as the issue)


I extracted it into a small script:
```bash
#!/bin/zsh

set -eu

prefix="${1:-${CONDA_PREFIX:-}}"
[[ -n "$prefix" ]] || {
  echo "usage: $0 [conda-prefix]" >&2
  exit 2
}

tmp="$(mktemp -d "${TMPDIR:-/tmp}/check-pari-version")"
src="$tmp/testfile.c"
out="$tmp/out.o"

cat > "$src" <<'EOF'
#include <stdio.h>
#include <pari/pari.h>
int main(void) {
  pari_init(1000000, 2);
  GEN v = pari_version(), M = gel(v,1), m = gel(v,2), p = gel(v,3);
  printf("%ld.%ld.%ld\n", itos(M), itos(m), itos(p));
  pari_close();
  return 0;
}
EOF

exec "${CC:-arm64-apple-darwin20.0.0-clang}" \
  -L"$prefix/lib" "$src" -o "$out" \
  -ftree-vectorize -fPIC -fstack-protector-strong -O2 -pipe \
  -isystem "$prefix/include" -D_FORTIFY_SOURCE=2 -isystem "$prefix/include" \
  -O0 -Werror=implicit-function-declaration -std=c17 \
  -lpari \
  -Wl,-headerpad_max_install_names \
  -Wl,-dead_strip_dylibs \
  -Wl,-rpath,"$prefix/lib"
```

On my machine this fails with:

```text
Undefined symbols for architecture arm64:
  "_printf", referenced from:
      _main in testfile-af924b.o
ld: symbol(s) not found for architecture arm64
arm64-apple-darwin20.0: error: linker command failed with exit code 1
```

so I think the main issue is the probe code in `src/meson.build : cc.run()....`. 
Maybe not relying on `printf`, just parse the version then check each part will be a good idea.

For example,
```c
#include <pari/pari.h>
int main(void) {
      pari_init(1000000, 2);
      GEN v = pari_version();
      long M = itos(gel(v, 1));
      long m = itos(gel(v, 2));
      long p = itos(gel(v, 3));
      pari_close();

      if (M > 2) return 0;
      if (M < 2) return 1;

      if (m > 17) return 0;
      if (m < 17) return 1;

      return p > 0 ? 0 : 1;
}
```

--- a/src/meson.build
+++ b/src/meson.build
@@ -89,12 +89,11 @@ except Exception:
   if pari.found() and not meson.is_cross_build()
     # Verify PARI version
     pari_version_code = '''
-    #include <stdio.h>
     #include <pari/pari.h>
     int main(void) {
       pari_init(1000000, 2);
       GEN v = pari_version(), M = gel(v,1), m = gel(v,2), p = gel(v,3);
-      printf("%ld.%ld.%ld", itos(M), itos(m), itos(p));
+      pari_printf("%ld.%ld.%ld", itos(M), itos(m), itos(p));
       pari_close();
       return 0;
     }

